core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
I agree that it's a flawed implementation, but it's reasonable to say they exist without causing problems today because split planning isn't done in parallel. I think your point about how someone would find this setting is valid, but I'm more concerned that someone that hits the problem should have a way around it. Without a way to turn this off, a user would be forced to either roll back their Spark version or fix and redeploy their InputFormat code. Both are expensive compared to a config option.